base/inc/array-utils.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * From this SO answer: https://stackoverflow.com/a/18781630/3710600
```

Comment:
I haven't checked this, but is code posted on Stack Overflow GPL compatible? Isn't their license creative commons or something? If it's not compatible, I assume this function would just need a rewrite.